data/adapters/depmap_adapter.py

```diff
@@ -0,0 +1,117 @@
+# A2M (2),0.014083962468692,0.0580835979659146,0.009988767666177,0.0065561956482438,0.0110209379651418,0.0060609112443749,0.0072958232853707,0.0179317372336054,0.0065760777739698
+# A2ML1 (144568),0.0338284534541587,0.0194828713741789,0.0087747320511599,0.0043219200079226,0.0091529955757067,0.0012808358556768,0.0047749277968025,0.0060950766332704,0.0044051485846711
+# A3GALT2 (127550),0.0495109597139512,0.0497933372469599,0.0993215225089184,0.0222002915981626,0.0216318159720975,0.0691636263860943,0.0171814961410507,0.0878640626935068,0.0289686576103241
+# A4GALT (53947),0.0049549043798291,0.0644720329570723,0.0995723644759633,0.0171211497280876,0.0677580998736281,0.1066244413858098,0.0842259302575638,0.0347167502889501,0.0467279313668812
+# A4GNT (51146),0.0269933824143704,0.0017749187371471,0.0075438544746256,0.0096048506255836,0.0135592402503593,0.0101073432171285,0.0021596303931416,0.0229722081510304,0.0652710662490125
+# AAAS (8086),0.131006482326207,0.0712890616055635,0.0496013615174664,0.1493775190934886,0.2167959119007355,0.0509210708384526,0.1354858251710864,0.1042089421935019,0.0645025243815082
+# AACS (65985),0.0031017110004655,0.0037316464915786,0.0475549666425148,0.052872923167593,0.0136603726345243,0.0079949689832072,0.0193872501805128,0.0305006545923165,0.0792603149045092
+
+# Other files needed for loading:
+# DepMap_model.csv is also downloaded from DepMap portal: https://depmap.org/portal/download/all/ in DepMap Public 23Q2 Primary Files set (Model.csv).
+# DepMap_gene_id_mapping.tsv is premapped file from gene symbol to gene ensembl id, queried from IGVF catalog gene collection.
+
+
+class DepMap(Adapter):
+    SOURCE = 'DepMap'
+    SOURCE_URL = 'https://depmap.org/portal/'
+    GENE_ID_MAPPING_PATH = './data_loading_support_files/DepMap_gene_id_mapping.tsv'
+    CELL_ONTOLOGY_ID_MAPPING_PATH = './data_loading_support_files/DepMap_model.csv'
+
+    CUTOFF = 0.5  # only load genes with dependency scores greater or equal to 0.5 for each cell
+
+    def __init__(self, filepath, type, label):
+        self.filepath = filepath
+        self.dataset = label
+        self.label = label
+        self.type = type
+
+        super(DepMap, self).__init__()
+
+    def process_file(self):
+        self.load_cell_ontology_id_mapping()
+        self.load_gene_id_mapping()
+
+        with open(self.filepath, 'r') as depmap_file:
+            # read model ids on the first row
+            model_ids = next(depmap_file).strip().split(',')[1:]
+            # record the column index of each model ID (start from 2nd column)
+            model_ids_column_mapping = {}
+            for column_index, model_id in enumerate(model_ids):
+                model_ids_column_mapping[column_index] = model_id
+                # check CVCL id mapping for all models once first
+                cell_ontology_id = self.cell_ontology_id_mapping[model_id]['cell_ontology_id']
```

Comment:
This check seems to be redundant.  You also checked it again later at line 73.

Reply:
Yes it's a bit redundant. I wanted to check them and output all cells missing ids at once first, before iterating over all genes (and getting missing ids message for every gene in that cell), if that makes sense.